platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/configurationcache/problems/ConfigurationCacheProblems.kt

```diff
@@ -145,17 +154,43 @@ class ConfigurationCacheProblems(
         }
     }
 
+    private
+    fun reportIncompatibleTask(path: String, reason: String) {
+        val problem = problemFactory
+            .problem {
+                text("task ")
+                reference(path)
+                text(" is incompatible with the configuration cache. ")
+                text("Reason: ")
+                text(reason)
+                text(".")
+            }
+            .mapLocation {
+                PropertyTrace.TaskPath(path)
+            }
+            .documentationSection(DocumentationSection.TaskOptOut).build()
+        onIncompatibleTask(problem)
+    }
+
     override fun onProblem(problem: PropertyProblem) {
         onProblem(problem, ProblemSeverity.Failure)
     }
 
     private
     fun onProblem(problem: PropertyProblem, severity: ProblemSeverity) {
         if (summarizer.onProblem(problem, severity)) {
-            report.onProblem(problem)
+            when (severity) {
+                ProblemSeverity.Warning -> report.onIncompatibleTask(problem)
```

Comment:
❔ It feels like there should be a more specific mechanism to distinguish between incompatible task problems. Not all warnings are necessarily incompatible task things
Something like `DiagnosticKind` seems a better fit here.

Reply:
Good point. This change was reverted, at some point it seemed necessary, but with the current state of this PR that does not seem required any longer